msgpack/fallback.py

```diff
@@ -246,16 +277,13 @@ def __init__(self, file_like=None, read_size=0, use_list=True,
             raise TypeError("`ext_hook` is not callable")
 
     def feed(self, next_bytes):
-        if isinstance(next_bytes, array.array):
-            next_bytes = next_bytes.tostring()
-        elif isinstance(next_bytes, bytearray):
-            next_bytes = bytes(next_bytes)
         assert self._fb_feeding
-        if (self._fb_buf_n + len(next_bytes) - self._fb_sloppiness
-                        > self._max_buffer_size):
+        view = _get_data_from_buffer(next_bytes, 1)
+        L = len(view)
+        if self._fb_buf_n + L - self._fb_sloppiness > self._max_buffer_size:
             raise BufferFull
-        self._fb_buf_n += len(next_bytes)
-        self._fb_buffers.append(next_bytes)
+        self._fb_buf_n += L
+        self._fb_buffers.append(view)
```

Comment:
Do copy here.
Supporting memoryview makes it possible to avoid memory copy on user side, not here.

```py
>>> import io
>>> f = io.BytesIO(b"foobar")
>>> buf = bytearray(4)
>>> n = f.readinto(buf)
>>>  n
4
>>> buf
bytearray(b'foob')
>>> a = [memoryview(buf)[:n]]
>>> n = f.readinto(buf)
>>> n
2
>>> buf
bytearray(b'arob')
>>> a += [memoryview(buf)[:n]]
>>> a
[<memory at 0x10ee60288>, <memory at 0x10ee60348>]
>>> bytes(a[0])
b'arob'   # Overwritten!!!
```